enhancements/results-forwarding.md

```diff
@@ -0,0 +1,504 @@
+recommendations if the supplied forwarding endpoint is valid. The Compliance
+Operator will also continue to use CRDs and PVCs to store results,
+remediations, and evidence as it does today.
+
+A user may explicitly disable in-cluster storage if forwarding is enabled. The
+Compliance Operator will not support disabling PVC storage without a valid
+forwarding endpoint.
+
+This will require changing the **aggregator** to always forward via a gRPC API
+to a configured implementation. In this case, the **aggregator** will act as a
+gRPC client. The **resultserver** then becomes a gRPC server, subject to the
+evidence forwarding provider and will be renamed **evidence-persistor** and
+responsible for writing results.
+
+Forwarder implementations would be configured via the `ScanSetting` and they'd be called "providers".
+For backwards compatibility, we'd have defaults that would point to the current mode of operation
+of the Compliance Operator. e.g. a `pvc` provider for evidence storage.
+
+For more detail on the proposed API, see the *API Extensions* section.
+
+### User Stories
+
+* As an Site Reliability Engineer managing multiple clusters, I'd like to run
+  compliance scans on my fleet in such a way that uses the minimal amount of
+  resources.
+
+* As a Site Reliability Engineer managing multiple clusters, I'd like to have
+  one place to view the compliance stance of my whole fleet.
+
+### API Extensions
+
+Given the scope of the change, we'd upgrade the `ScanSetting` version to
+`v1alpha2`. The following sections describe the additional values needed by the
+`ScanSetting` CRD.
+
+#### `resultForwarding.provider`
+
+Required: True
+Type: string
+
+The implementation to use for forwarding. The initial implementation will
+support `grpc`.
+
+#### `resultForwarding.grpc.endpoint`
+
+Required: True
+Type: string
+
+Endpoint URL to a gRPC server for result storage. The service may be in or
+outside the cluster. One potential implementation may be
+[compserv](https://github.com/rhmdnd/compserv).
+
+#### `resultForwarding.grpc.version`
+
+Required: False
+Default: `v1`
+Type: string
+
+#### `resultForwarding.grpc.tls.dynamic.caSecretName`
+
+Required: True
+Type: string
+
+Reference to a CA certificate and key pair for TLS.
+
+#### `resultForwarding.grpc.tls.dynamic.serverSecretName`
+
+Required: True
+Type: string
+
+Reference to a server certificate/keypair secret for mutual TLS.
+
+#### `resultForwarding.grpc.tls.dynamic.clientSecretName`
+
+Required: True if using mutual TLS
+Type: string
+
+Reference to a client certificate/keypair secret for mutual TLS.
+
+#### `resultForwarding.grpc.authentication.type`
+
+Required: False
+Default: `mtls`
+Type: string
+Choices: `mtls`, `token`
+
+The type of authentication the Compliance Operator should use to authenticate
+the forwarding endpoint.
+
+#### `resultForwarding.grpc.authentication.token`
+
+Required: True if using `token` authentication type
+Type: string
```

Comment:
Do I understand it correctly that the `resultForwarding.grpc.authentication.token` field contains the same thing directly in the setting that `resultForwarding.grpc.authentication.token.tokenSecretName` in a secret?
Is there a reason to not only have the secret?

Reply:
uhm... yeah, this is confusing. `token` shouldn't accept strings. Instead, it should be an object (dict?) that allows you to set the token-related configurations.

Reply:
Ok, thanks for the clarification. I misunderstood this.

Do you have an example of something that already uses this approach?

Reply:
Something like our OutputRef: https://github.com/ComplianceAsCode/compliance-operator/blob/f0212eaf2ed5fa9e0e7917cdda80219f6af7a383/pkg/apis/compliance/v1alpha1/tailoredprofile_types.go#L80
Even if there's just a Name field and we always assume the operator namespace, it's still a good idea to define it like that.

Reply:
Ok - got it. That makes a little more sense.

So this should be removed and instead we would have `token.secretName` and `token.secretNamespace`, allowing the compliance operator to fetch the token.